tests/Feature/Flights/StoreFlightTest.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\Feature\Flights;
+
+use Carbon\CarbonImmutable;
+use Database\Factories\AirlineFactory;
+use Database\Factories\CityFactory;
+use Illuminate\Testing\Fluent\AssertableJson;
+use function Pest\Laravel\assertDatabaseHas;
+use function Pest\Laravel\postJson;
+
+describe('flights', function (): void {
+    it('can create a flight successfully', function (): void {
+        $airline = AirlineFactory::new()->createOne();
+        $departureCity = CityFactory::new()->createOne();
+        $arrivalCity = CityFactory::new()->createOne();
+
+        $data = [
+            'airline_id' => $airline->id,
+            'departure_city_id' => $departureCity->id,
+            'arrival_city_id' => $arrivalCity->id,
+            'departure_time' => now()->addDay()->format('Y-m-d H:i:s'),
+            'arrival_time' => now()->addDays(10)->format('Y-m-d H:i:s'),
+        ];
+
+        $response = postJson('/api/flights', $data);
+
+        $response
+            ->assertCreated()
+            ->assertJson(
+                fn (AssertableJson $json): AssertableJson =>
+                $json->has(
+                    'data',
+                    fn (AssertableJson $json): AssertableJson =>
+                    $json
+                        ->where('airline_id', $airline->id)
+                        ->where('departure_city_id', $departureCity->id)
+                        ->where('arrival_city_id', $arrivalCity->id)
+                        ->where('departure_time', CarbonImmutable::parse($data['departure_time'])->toJSON())
+                        ->where('arrival_time', CarbonImmutable::parse($data['arrival_time'])->toJSON())
+                        ->etc()
+                )
+            );
+        assertDatabaseHas('flights', [
+            'airline_id' => $airline->id,
+            'departure_city_id' => $departureCity->id,
+            'arrival_city_id' => $arrivalCity->id,
+            'departure_time' => $data['departure_time'],
+            'arrival_time' => $data['arrival_time'],
+        ]);
+    });
+
+    it('cannot create a flight with invalid data', function (string $field, mixed $value): void {
+        $airline = AirlineFactory::new()->createOne();
+        $departureCity = CityFactory::new()->createOne();
+        $arrivalCity = CityFactory::new()->createOne();
+
+        $data = [
+            'airline_id' => $airline->id,
+            'departure_city_id' => $departureCity->id,
+            'arrival_city_id' => $arrivalCity->id,
+            'departure_time' => now()->addDay()->format('Y-m-d H:i:s'),
+            'arrival_time' => now()->addDays(2)->format('Y-m-d H:i:s'),
+        ];
+
+        $data[$field] = $value;
+
+        $response = postJson('/api/flights', $data);
+
+        $response->assertUnprocessable()
+            ->assertJsonValidationErrors([$field], 'error.fields');
+    })->with([
```

Comment:
the only thing you didn't check was that the arrival time should be after departure time, can you add that one?

Reply:
same for update controller